index.d.ts

```diff
@@ -63,3 +63,49 @@ console.log('done');
 */
 export function writePackageSync(path: string, data: JsonObject, options?: Options): void;
 export function writePackageSync(data: JsonObject, options?: Options): void;
+
+// TODO: make `data` use PackageJson type?
+
+/**
+Update a `package.json` file.
+
+Writes atomically and creates directories for you as needed. Sorts dependencies when writing. Preserves the indentation if the file already exists.
+
+@param path - The path to where the `package.json` file should be written or its directory.
+
+@example
+```
+import path from 'node:path';
+import {updatePackage} from 'write-pkg';
+
+await updatePackage({foo: true});
+//=> { "foo": true }
+
+await updatePackage({foo: false, bar: true});
+//=> { "foo": false, "bar": true }
+```
+*/
+export function updatePackage(path: string, data: JsonObject, options?: Options): Promise<void>;
+export function updatePackage(data: JsonObject, options?: Options): Promise<void>;
```

Comment:
Should data be [`JsonObject`](https://github.com/sindresorhus/type-fest/blob/a83ff3a80e7fa19374d7d7e66d9269886cf9e244/source/basic.d.ts#L38-L45) | [`PackageJson`](https://github.com/sindresorhus/type-fest/blob/main/source/package-json.d.ts)? That would still allow arbitrary fields, but add completions for `package.json` properties.

Reply:
Sure. I would do `PackageJson | JsonObject` as it's generally best to place the most specific one first.